sample/client/oauth.py

```diff
@@ -0,0 +1,217 @@
+#
+
+import hashlib
+import binascii
+import evernote.edam.userstore.constants as UserStoreConstants
+import evernote.edam.type.ttypes as Types
+import evernote.edam.notestore.ttypes as NoteStoreTypes
+import sys
+
+from evernote.api.client import EvernoteClient
+
+# Real applications authenticate with Evernote using OAuth
+# to get an API key and secret visit
+# https://dev.evernote.com/#apikey
+CONSUMER_KEY = "INPUT CONSUMER KEY HERE"
+CONSUMER_SECRET = "INPUT CONSUMER SECRET HERE"
+sandbox = True #if True will use sandbox.evernote.com, if False will use www.evenrote.com
+
+#if sandbox True will use sandbox.evernote.com, if False will use www.evenrote.com
+if sandbox:
+    EN_URL="https://sandbox.evernote.com"
+else:
+    EN_URL="https://www.evernote.com"
+
+#Throw an error and exit if there is no Consumer key or secret entered
+# to get a key go to https://dev.evernote.com#apikey
+if CONSUMER_KEY=="PUT API KEY HERE" or CONSUMER_KEY=="" or CONSUMER_SECRET=="PUT API SECRET HERE" or CONSUMER_SECRET=="":
+    print """ERROR: Edit the server.py file and add your consumer key and consumer secret in lines 24 and 25 at the begining of "oauth.py".
+    \nIf you do not have a Evernote consumer key and secret go to https://dev.evernote.com#apikey to get one (for free!).\n"""
+    sys.exit(1)
+
+
+# Initial development is performed on our sandbox server. To use the production
+# service, change sandbox=False and request your API be approved for access 
+# on Evernote's production service 
+
+#setup client
+client = EvernoteClient(
+consumer_key=CONSUMER_KEY,
+consumer_secret=CONSUMER_SECRET,
+sandbox= sandbox
+)
+
+request_token = client.get_request_token("http://evernote") #set callback URL
+try:
+    oauth_token = request_token['oauth_token'] #set temporary oauth token
+    oauth_token_secret = request_token['oauth_token_secret'] # set temporary oaut token secret
+    authorize_url = client.get_authorize_url(request_token) #get redirect URL
+except KeyError:
+    print "Incorrect consumer key or secret.  Please enter a valid consumer key and secret and try again.  \n\nPlease note that key must be approved to be used on www.evernote.com while all new keys are active by default on sandbox.evernote.com\n\n"
+    sys.exit(1)    
+
+
+print "Please complete the following steps to continue: "
+print "(1) Go to: " + authorize_url
+print "(2) Grant access to your application"
+print "(3) Look at the URL you are directed to and find the string after '&oauth_verifier=' and before '&sandbox_lnb'"
+oauth_verifier = raw_input("(4) Enter that string (the OAuth verifier) here: ")
+
+auth_token=None
+
+#get access token
+while not auth_token:
+    try:
+        auth_token = client.get_access_token(oauth_token, oauth_token_secret, oauth_verifier)
+    except:
+        oauth_verifier = raw_input("Incorrect OAuth verifier.\nTry again or press control+Z to exit.\nOAuthVerifier: ")        
+
+user_store = client.get_user_store()
```

Comment:
Consider factoring the code shared by this an `dev_token.py` into a common module.